python_pkgs/sh_expand/expand.py

```diff
@@ -0,0 +1,518 @@
+import copy
```

Comment:
This is a different package (https://github.com/binpash/sh-expand)! and the changes in this file need to happen in a PR there.

Reply:
> Would you recommend for this:
> 
>     1. keeping the ExpansionError in the custom_error and importing it here, or
> 
>     2. moving the ExpansionError here and importing it to pash_compiler to catch it?

I would probably define ExpansionError in sh-expand, and then import it in pash to catch it. It makes sense to hide all expansion related behavior and details in that package.

> I'm not sure why, but it also seems like I don't have permission to push my branch onto this package repo.

That should be fixed now, could you try again? :)

Reply:
As you suggested, I imported `ExpansionError` into pash_compiler and defined `ExpansionError` in sh_expand in the original package (with a new [PR](https://github.com/binpash/sh-expand/pull/5))! 

I think currently the tests on github actions are not passing due to `ExpansionError` not being found in the package yet. I've ran the tests locally with the most recent changes, and it seems to work as before so hopefully after the change in the package the issue of the `ExpansionError` not found in the import will not be a problem. 

Thank you!

Reply:
I merged that PR (after bumping the project version because that needs to happen for the package to be properly deployed to PyPI). 

Now the tests seem to be passing, so the only thing you need to do is sign off your commits (follow the instructions here) https://github.com/binpash/pash/pull/721/checks?check_run_id=26107013047 and then we can properly approve and merge!

Great work :)